pkg/datapath/iptables/iptables.go

```diff
@@ -929,147 +1084,39 @@ func (m *IptablesManager) InstallRules(ifName string) error {
-			//     range
-			// * Non-tunnel mode:
-			//   * May not be targeted to an IP in the cluster range
-			progArgs := append(
-				m.waitArgs,
-				"-t", "nat",
-				"-A", ciliumPostNatChain,
-				"!", "-d", datapath.RemoteSNATDstAddrExclusionCIDR().String(),
-			)
-			if option.Config.EgressMasqueradeInterfaces != "" {
-				progArgs = append(
-					progArgs,
-					"-o", option.Config.EgressMasqueradeInterfaces)
-			} else {
-				progArgs = append(
-					progArgs,
-					"-s", node.GetIPv4AllocRange().String(),
-					"!", "-o", "cilium_+")
-			}
-			progArgs = append(
-				progArgs,
-				"-m", "comment", "--comment", "cilium masquerade non-cluster",
-				"-j", "MASQUERADE")
-			if option.Config.IPTablesRandomFully {
-				progArgs = append(progArgs, "--random-fully")
-			}
-			if err := runProg("iptables", progArgs, false); err != nil {
-				return err
-			}
-
-			// The following rules exclude traffic from the remaining rules in this chain.
-			// If any of these rules match, none of the remaining rules in this chain
-			// are considered.
-			// Exclude traffic for other than interface from the masquarade rules.
-			// RETURN fro the chain as it is possible that other rules need to be matched.
-			if err := runProg("iptables", append(
-				m.waitArgs,
-				"-t", "nat",
-				"-A", ciliumPostNatChain,
-				"!", "-o", localDeliveryInterface,
-				"-m", "comment", "--comment", "exclude non-"+ifName+" traffic from masquerade",
-				"-j", "RETURN"), false); err != nil {
-				return err
-			}
-
-			// Exclude proxy return traffic from the masquarade rules
-			if err := runProg("iptables", append(
-				m.waitArgs,
-				"-t", "nat",
-				"-A", ciliumPostNatChain,
-				"-m", "mark", "--mark", matchFromProxy, // Don't match proxy (return) traffic
-				"-m", "comment", "--comment", "exclude proxy return traffic from masquarade",
-				"-j", "ACCEPT"), false); err != nil {
-				return err
-			}
-
-			if option.Config.Tunnel != option.TunnelDisabled {
-				// Masquerade all traffic from the host into the ifName
-				// interface if the source is not the internal IP
-				//
-				// The following conditions must be met:
-				// * Must be targeted for the ifName interface
-				// * Must be targeted to an IP that is not local
-				// * Tunnel mode:
-				//   * May not already be originating from the masquerade IP
-				// * Non-tunnel mode:
-				//   * May not orignate from any IP inside of the cluster range
-				if err := runProg("iptables", append(
-					m.waitArgs,
-					"-t", "nat",
-					"-A", ciliumPostNatChain,
-					"!", "-s", node.GetHostMasqueradeIPv4().String(),
-					"!", "-d", node.GetIPv4AllocRange().String(),
-					"-o", "cilium_host",
-					"-m", "comment", "--comment", "cilium host->cluster masquerade",
-					"-j", "SNAT", "--to-source", node.GetHostMasqueradeIPv4().String()), false); err != nil {
-					return err
-				}
-			}
+		if err := m.installMasqueradeRules("iptables", ifName, localDeliveryInterface,
```

Comment:
To make both cases symmetric, maybe we should wrap the invocation into `if option.Config.EnableIPv4 { ... }`?

Reply:
This already seems to be wrapped inside if `option.Config.EnableIPv4{ ... }`
https://github.com/cilium/cilium/pull/14124/files#diff-465a6b3f4ab9e1e94fe39e5aefe81c41eaa1dc99c66500161072e564029c2f51R1052